src/components/FeatureViews/Registration/CourseList.js

```diff
@@ -59,45 +63,81 @@ const CourseList = ({ filteredCourses, updatedParent }) => {
-        "variables": { "userIds": studentList },
+
+    const {currentParent, ...registrationCartState} = useSelector((state) => state.Registration);
+    const dispatch = useDispatch();
+
+    const {studentList} = JSON.parse(sessionStorage.getItem("registrations"))?.currentParent || false;
+    const {data, loading} = useQuery(GET_STUDENTS_AND_ENROLLMENTS, {
+        "variables": {"userIds": studentList},
         skip: !studentList
     });
 
-    const { currentParent } = getRegistrationCart();
-    const { parentIsLoggedIn } = useValidateRegisteringParent();
-    const { courseTitle, courseRow } = useStyles();
+    const {parentIsLoggedIn} = useValidateRegisteringParent();
+    const {courseTitle, courseRow} = useStyles();
+
+    if (loading) return <Loading small/>;
 
-    if (loading) return <Loading small />;
+    const validRegistrations = Object.values(registrationCartState)
+        .filter(registration => registration);
+    const registrations = validRegistrations && [].concat.apply([], validRegistrations);
+    const studentOptions = data?.userInfos
+        .filter(({user}) => (!registrations.find(({course, student}) =>
+                (course.id === quickCourseID && user.id === student))
+        ))
+        .map((student) => ({
+            "label": fullName(student.user),
+            "value": student.user.id,
+        })) || [];
 
-    const studentOptions = data?.userInfos.map((student) => ({
-        "label": fullName(student.user),
-        "value": student.user.id,
-    })) || [];
+    const enrolledCourseIds = data?.enrollments.map(({course}) => course.id);
+    const previouslyEnrolled = (courseId, enrolledCourseIds, registrations, studentList) => {
+        const validRegistrations = Object.values(registrations)
+            .filter(registration => registration);
+        if (!studentList || validRegistrations.length === 0 || !enrolledCourseIds) return false;
+        const registeredCourseIds = registrations.map(({course}) => course.id);
+        const numStudents = studentList.length;
+        const countOccurrences = (arr, val) => arr.reduce((a, v) => (v === val ? a + 1 : a), 0);
 
-    const handleStartQuickRegister = (courseID) => () => {
+        return !(countOccurrences(registeredCourseIds, courseId) < numStudents &&
+            !enrolledCourseIds.includes(courseId));
+    }
+
+    const handleStartQuickRegister = (courseID) => (e) => {
+        e.preventDefault();
         setOpen(true);
         setQuickCourseID(courseID);
     };
 
     const handleAddRegistration = () => {
-        submitRegistration(quickStudent, quickCourseID);
+        dispatch({
+            type: types.ADD_CLASS_REGISTRATION,
+            payload: {
+                studentId: quickStudent,
+                courseId: quickCourseID
+            }
+        });
         setOpen(false);
     }
 
+    const disableQuickRegister = ({course, enrolledCourseIds, registrations, studentList}) => {
```

Comment:
Rename to `shouldDisableRegister` to clarify that this returns a boolean instead of performing an action.